### PR TITLE
[WIP] Variants

### PR DESCRIPTION
**Pull request by hajimehoshi**
The implementation is not finished yet. Not all tests pass.

I used `header.ParseValueAndParams` as the counterpart of `ParseParameterisedList`, but I feel like I'd need to implement the counterpart myself.

### CODE REVIEW

**Comment by hajimehoshi** [comment]
This PR just implements `variants.MatchRequest`. Let's discuss how to implement actual usages later.

PTAL

**Comment by hajimehoshi** [comment]
Thanks. It looks like I'd need to implement `MatchesMimeType` in Go. I couldn't find a third-party library that does the same thing.

File: go/signedexchange/internal/variants/variants.go [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+
+	"github.com/golang/gddo/httputil/header"
+)
+
+func MatchesMimeType(mimeTypePattern, mimeType string) bool {
+	// TODO(hajimehoshi): Implement this. See Chromium's MatchesMimeType.
+	return mimeTypePattern == mimeType
+}
+
+func ParseListOfLists(h http.Header, key string) [][]string {
```
Comment (irori) [inline go/signedexchange/internal/variants/variants.go]: Please use `structuredheader.ParseListOfLists` (added at #395)

**Comment by hayatoito** [comment]
Closing.

**Pull request closed without merging**